### ISSUE TITLE
Better handling of HTML to XHTML

### ISSUE THREAD

**Issue by dteviot** [dteviot/WebToEpub#118]
There are a number of HTML pages that are not correctly converted to XHTML.
e.g. 
- http://skythewood.blogspot.com/ has &lt;o:p&gt; nodes. I think they're supposed to be &lt;p&gt; nodes, but there's no 'o' namespace
- http://novelplanet.com/Novel/10-nen-goshi-no-HikiNiito-o-Yamete-Gaishutsushitara-Jitaku-goto-Isekai-ni-Ten-ishiteta/c6-part9?id=195323

It's also annoying that user is only aware of problem when the EPUB reader faults on the page.
Fix probably needs to include following 
- Have plug-in validate the generated XHTML and warn when there's a problem.
- Come up with more reliable way to convert.  Note, short term fix may be examine the failure cases and see if there's a common issue that can be fixed, before writing own HTML to XHTML converter.

**Comment by toshiya44** [comment]
Yep. They're supposed to just `<p>` nodes. Apparently MS word does that when a doc is converted to html.
https://stackoverflow.com/questions/7808968/what-do-op-elements-do-anyway

**Comment by dteviot** [comment]
@typhoon71, @toshiya44, @dreamer2908

Latest commit to Experimental Tab Branch should now generate EPUB v3 files if you check the "Create EPUB 3" advanced option.
I hope this will solve the problem with sometimes not being able to convert HTML into valid XHTML.
(EPUB 3 uses HTML 5 instead of XHTML.)
This requires your EPUB reader to support EPUB 3, but hopefully by now most do.
Please try it out and let me know how well the EPUB 3 works with your readers.
Thanks.

**Comment by toshiya44** [comment]
To be honest I'm not very knowledgeable about EPUB3. I'm using [this Sigil plugin](https://www.mobileread.com/forums/showthread.php?s=1f4fb04ba0b183f43db7c975c1d5f230&t=248186), which is supposed to contain [EpubCheck 4.0.2](https://github.com/IDPF/epubcheck), in order to check errors.

Calibre officially doesn't endorse EPUB3 due to various reasons, so the editor part of Calibre is not very specialized for this. Please see [this thread](https://www.mobileread.com/forums/showthread.php?s=06ae86340a8b61dd23a788eb3493a88e&t=260467&page=4). However, the viewer has no issues with rendering EPUB3 (this was also discussed in that thread). 

---------------

**URL**: https://royalroadl.com/fiction/5288/how-to-avoid-death-on-a-daily-basis/chapter/69854/73-night-of-the-living-zombers

**Errors**: In the OPF, the image is listed as `"image/jpeg"` even though it's a has a PNG extension (apparently the image is actually a bmp, no clue what's going on). 
There's also an error notice for the "Form Feed" (value 0x0c) character that you mentioned in the issue. 

**URL**: https://skythewood.blogspot.ca/2017/07/F15.html

**Error**: Epubcheck complains about having a `name` attribute ( `<a name="more"></a>` ).

By the way, According to [this stackoverflow thread](https://stackoverflow.com/questions/1397592/difference-between-id-and-name-attributes-in-html), there doesn't seem to be any difference between `name` and `id` attribute in the context of ePub. So wouldn't it be fine to replace `name` with `id`? I've seen `name` attributes used as `id`s in wordpress sites as well.

**Comment by dteviot** [comment]
@toshiya44 
Firstly, thanks very much for the prompt response.

> Errors: In the OPF, the image is listed as "image/jpeg" even though it's a has a PNG extension (apparently the image is actually a bmp, no clue what's going on).

I assume you're referring to: https://cdn.royalroadl.com/mooderino/6edd9796-3cb7-434c-a5ad-bb7dece2967a.png.  It's listed as "image/jpeg" in the OPF is because when the file is fetched from the web server, the "content-type" in the HTTP response was "image/jpeg", so that's what went into the OPF file.  Images don't always have extensions, so I was relying on the content-type.  That said, this doesn't seem to cause any problems with the reader.

> There's also an error notice for the "Form Feed" (value 0x0c) character that you mentioned in the issue.

Um.  yes, I haven't fixed the warnings yet.  That said, with EPUB viewer, it would not show the chapter due to the Form Feed character.  Now it shows the HTML without a problem.
